### PR TITLE
Glossary

### PR DESCRIPTION
**Pull request by tonyoconnell**
#### Description (required)

Added 71 terms to the glossary. Total items now 73

#### Related issues & labels (optional)

- Closes #
- Suggested label:

### CODE REVIEW

**Comment by at-the-vr** [comment]
it usually helps maintainers if you can edit the title more appropriately example https://github.com/withastro/docs/pull/5529 (you dont have to be specific like genteure did but give it a quick change for faster response)

**Review by at-the-vr — COMMENTED** [review]
I see you have also mentioned changes for [Using Bun with Astro](https://docs.astro.build/en/recipes/bun/) and its not huge so maintainers won't have trouble dealing with it.

Its generally better to stick to the topic in the PR (that is `update glossary.mdx` in this case) and create a separate PR for `update bun.mdx` from next time

**Comment by sarah11918** [comment]
(PS - As per the rules, I don't merge a glossary PR until I see some kind of approval and generalized consensus that you all want this merged!)

**Comment by sarah11918** [comment]
There we go! Updating main into this branch fixed it. 

Remember, with glossary branch comes glossary responsibility!  :smile:

**Comment by sarah11918** [comment]
@at-the-vr I didn't know whether you wanted them worked in, or separate since they'd been discussed and merged in separately... kind of like keeping the list at the bottom as "potential candidates to pull from." But yes, feel free! It's your glossary!

File: contributor-guides/glossary.md [hunk 1]
```diff
@@ -7,3 +7,518 @@ Astro components are the basic building blocks of any Astro project. They are HT
...
+-----
+
+SARAH RESOLVING CONFLICTS
+Additions from this PR are added below the existing content
+
+-----
+
+## Adaptors
+
+Adaptors are essential for deploying Astro projects to different hosting platforms. Each adaptor is tailored to a specific host, such as Netlify, Vercel, or Cloudflare Pages, ensuring compatibility and streamlined deployment processes. Adaptors handle the nuances of each platform, providing a seamless integration for Astro projects.
```
Comment (VoxelMC) [inline contributor-guides/glossary.md:20]:
```suggestion
## Adapters

Adapters are essential for deploying Astro projects to different hosting platforms. Each adapter is tailored to a specific host, such as Netlify, Vercel, or Cloudflare Pages, ensuring compatibility and streamlined deployment processes. Adapters handle the nuances of each platform, providing a seamless integration for Astro projects.
```

Just some typos :)

File: contributor-guides/glossary.md [hunk 2]
```diff
@@ -7,3 +7,518 @@ Astro components are the basic building blocks of any Astro project. They are HT
...
+
+For more details, see:
+- [Astro Adaptor API Reference](https://docs.astro.build/en/reference/adapter-reference/)
```
Comment (VoxelMC) [inline contributor-guides/glossary.md:23]:
```suggestion
- [Astro Adapter API Reference](https://docs.astro.build/en/reference/adapter-reference/)
```

File: contributor-guides/glossary.md [hunk 3]
```diff
@@ -7,3 +7,518 @@ Astro components are the basic building blocks of any Astro project. They are HT
...
+ 
+## API
```
Comment (VoxelMC) [inline contributor-guides/glossary.md:25]: Not a change, but just a note that we will need to classify these based on Astro-relatedness at some point in the future.

File: contributor-guides/glossary.md [hunk 4]
```diff
@@ -7,3 +7,518 @@ Astro components are the basic building blocks of any Astro project. They are HT
...
+- [Astro Build Command](https://docs.astro.build/en/reference/cli-reference/#astro-build)
+- [Deploy your Astro Site](https://docs.astro.build/en/guides/deploy/)
+
+## Client Side Rendering
+
+Client Side Rendering involves rendering web components in the browser through JavaScript. Unlike server-side rendering, client side rendering is used for dynamic content that updates in response to user interactions, providing a more interactive and responsive experience.
+
+For more details, see:
+- [Client-Side Scripts](https://docs.astro.build/en/guides/client-side-scripts/#client-side-scripts)
+- [Client Only Directive](https://docs.astro.build/en/reference/directives-reference/#clientonly)
```
Comment (VoxelMC) [inline contributor-guides/glossary.md:46]:
```suggestion
- [`client:only` Directive](https://docs.astro.build/en/reference/directives-reference/#clientonly)
```
To follow suit with the docs.

File: contributor-guides/glossary.md [hunk 5]
```diff
@@ -7,3 +7,518 @@ Astro components are the basic building blocks of any Astro project. They are HT
...
+
+## Cloudflare
+
+Astro's integration with Cloudflare allows for deploying projects on Cloudflare's edge network, enhancing site performance and reliability. This integration benefits from Cloudflare's global distribution, offering faster content delivery to users.
+
+For more details, see:
+- [Deploying on Cloudflare Guide](https://docs.astro.build/en/guides/deploy/cloudflare/)
```
Comment (VoxelMC) [inline contributor-guides/glossary.md:53]:
```suggestion
Astro's integration with Cloudflare allows for deploying projects to Cloudflare's edge network, enhancing site performance and reliability. This integration benefits from Cloudflare's global distribution, offering faster content delivery to users.

For more details, see:
- [Guide: Deploying to Cloudflare](https://docs.astro.build/en/guides/deploy/cloudflare/)
```

File: contributor-guides/glossary.md [hunk 6]
```diff
@@ -7,3 +7,518 @@ Astro components are the basic building blocks of any Astro project. They are HT
...
+For more details, see:
+- [Astro API Reference](https://docs.astro.build/en/reference/api-reference/)
+
+## Build
+
+Building refers to the process of compiling an Astro project into static files. This step transforms components, scripts, and styles into optimized HTML, CSS, and JavaScript, preparing them for deployment. The build process includes essential optimizations like minification and code splitting, crucial for enhancing performance.
+
+For more details, see:
+- [Astro Build Command](https://docs.astro.build/en/reference/cli-reference/#astro-build)
+- [Deploy your Astro Site](https://docs.astro.build/en/guides/deploy/)
```
Comment (VoxelMC) [inline contributor-guides/glossary.md:38]:
```suggestion
- [Guide: Deploy your Astro Site](https://docs.astro.build/en/guides/deploy/)
```

File: contributor-guides/glossary.md [hunk 7]
```diff
@@ -7,3 +7,518 @@ Astro components are the basic building blocks of any Astro project. They are HT
...
+## Cloudflare
+
+Astro's integration with Cloudflare allows for deploying projects on Cloudflare's edge network, enhancing site performance and reliability. This integration benefits from Cloudflare's global distribution, offering faster content delivery to users.
+
+For more details, see:
+- [Deploying on Cloudflare Guide](https://docs.astro.build/en/guides/deploy/cloudflare/)
+
+## Collection Entry
+
+A Collection Entry is a content piece within a collection directory, supporting formats like Markdown, MDX, YAML, and JSON. These entries are crucial for content organization and management projects.
```
Comment (VoxelMC) [inline contributor-guides/glossary.md:57]:
```suggestion
A Collection Entry is a content piece within a collection directory, supporting formats like Markdown, MDX, YAML, and JSON. 
```

This definition should probably be more in-depth. Content collections and their parts are pretty `crucial` to the Astro experience! Perhaps we can include more about how a collection entry is rendered and more about what a `collection directory` is (and perhaps a re-wording of this).

File: contributor-guides/glossary.md [hunk 8]
```diff
@@ -7,3 +7,518 @@ Astro components are the basic building blocks of any Astro project. They are HT
...
+
+For more details, see:
+- [Content Collections](https://docs.astro.build/en/guides/content-collections/)
+
+## Components
```
Comment (VoxelMC) [inline contributor-guides/glossary.md:62]:
```suggestion
## Component
```
I will ask others their opinion, but I think the terms should be singular - like a dictionary. LMK what you think of this.

File: contributor-guides/glossary.md [hunk 9]
```diff
@@ -7,3 +7,518 @@ Astro components are the basic building blocks of any Astro project. They are HT
...
+- [Configuration Reference](https://docs.astro.build/en/reference/configuration-reference/)
+
+## Content Collections
+
+Content Collections refer to a method for organizing and managing content within an Astro project. A content collection is a top-level directory inside the `src/content` directory, like `src/content/newsletter` or `src/content/authors`. Collection entries can be in various formats, including Markdown, MDX, YAML, and JSON. These collections are integral for structuring documents, validating frontmatter, and providing TypeScript type-safety.
+
+For more details, see:
+- [Content Collections](https://docs.astro.build/en/guides/content-collections/)
+
+## CSS
```
Comment (VoxelMC) [inline contributor-guides/glossary.md:84]: Perhaps change this to `Scoped Styles` in general
```suggestion
## Scoped Styles
```

File: contributor-guides/glossary.md [hunk 10]
```diff
@@ -7,3 +7,518 @@ Astro components are the basic building blocks of any Astro project. They are HT
...
+
+## Configuration
+
+Configuration involves setting up global project settings via the `astro.config.mjs` file. This includes defining build options, customizing the development environment, and integrating with various tools and frameworks.
+
+For more details, see:
+- [Configuring Astro Guide](https://docs.astro.build/en/guides/configuring-astro/)
+- [Configuration Reference](https://docs.astro.build/en/reference/configuration-reference/)
+
+## Content Collections
```
Comment (VoxelMC) [inline contributor-guides/glossary.md:77]:
```suggestion
## Content Collection
```

File: contributor-guides/glossary.md [hunk 11]
```diff
@@ -7,3 +7,518 @@ Astro components are the basic building blocks of any Astro project. They are HT
...
+
+Content Collections refer to a method for organizing and managing content within an Astro project. A content collection is a top-level directory inside the `src/content` directory, like `src/content/newsletter` or `src/content/authors`. Collection entries can be in various formats, including Markdown, MDX, YAML, and JSON. These collections are integral for structuring documents, validating frontmatter, and providing TypeScript type-safety.
```
Comment (VoxelMC) [inline contributor-guides/glossary.md:79]:
```suggestion
Content Collections are used to organize and manage content within an Astro project. A content collection is a top-level directory inside the `src/content` directory, like `src/content/newsletter` or `src/content/authors`. Collection entries can be in various formats, including Markdown, MDX, YAML, and JSON. These collections are integral for structuring documents, validating frontmatter, and providing TypeScript type-safety.
```

**Review by VoxelMC — CHANGES_REQUESTED** [review]
Here is my review of the first quarter of the file.

*In the future, can we only submit a handful of terms at once per PR?*
I am impressed with the amount of work it is, but it's a lot to read, and things might slip through the cracks!

The first quarter looks good; please review my comments and let me know what you think of them.

**Comment by sarah11918** [comment]
Hi everyone! I won't close this PR just yet, to let you grab whatever content and comments you want from it, but please see my comments in the Discord thread about how we'll proceed with this. This current system isn't working, and I've left a message there about what we'll do going forward as this gets off the ground.

So, please be prepared to snag whatever from this PR you want and incorporate it into the new direction, then we'll close this one!

**Review by sarah11918 — CHANGES_REQUESTED** [review]
This is to be closed once the content from here is incorporated in the new solution.

**Comment by sarah11918** [comment]
Hi @at-the-vr @VoxelMC ! I'm going to close this PR now, but of course you can still look at it for reference and snag stuff for HackMD!

**Pull request closed without merging**